src/data/fintype/basic.lean

```diff
@@ -1006,8 +1006,9 @@ begin
 end
 
 /-- If the subtype of all-but-one elements is a `fintype` then the type itself is a `fintype`. -/
-def fintype_of_fintype_ne (a : α) [decidable_pred (= a)] (h : fintype {b // b ≠ a}) : fintype α :=
-fintype.of_equiv _ $ equiv.sum_compl (= a)
+def fintype_of_fintype_ne (a : α) (h : fintype {b // b ≠ a}) : fintype α :=
+fintype.of_bijective (sum.elim (coe : {b // b = a} → α) (coe : {b // b ≠ a} → α)) $
+  by { classical, exact (equiv.sum_compl (= a)).bijective }
```

Comment:
Why is this new proof so much longer than the old one? Can't you just do?
```suggestion
by { classical, exact fintype.of_equiv _ $ equiv.sum_compl (= a) }
```

Reply:
With my proof, it is clear that the "data" part of the definition does not depend on `classical.choice`. I don't know if your definition is computable.

Reply:
This is just the old proof with `classical` slapped on top, so the data part should be computable as well.

Reply:
Your definition is noncomputable. My definition still compiles if I remove type annotations from `coe` but I'm not sure how much this will affect readability.